InventorySales/Plugin/Sales/OrderManagement/AppendReservationsAfterOrderPlacementPlugin.php

```diff
@@ -160,6 +181,27 @@ public function afterPlace(OrderManagementInterface $subject, OrderInterface $or
         ]);
 
         $this->placeReservationsForSalesEvent->execute($itemsToSell, $salesChannel, $salesEvent);
+
+        try {
+            $order = $proceed($order);
+        } catch (\Exception $e) {
+            //add compensation
+            foreach ($itemsToSell as $item) {
+                $item->setQuantity(-(float)$item->getQuantity());
+            }
+
+            /** @var SalesEventInterface $salesEvent */
+            $salesEvent = $this->salesEventFactory->create([
+                'type' => SalesEventInterface::EVENT_ORDER_CANCELED,
```

Comment:
It's not an order cancellation event. 
I would probably introduce another event type, like order_place_failed

Reply:
Added new constant here, but this makes this fix bic, so probably it will be delivered only to 1.2